### ISSUE TITLE
Milestone 2 requirements

### ISSUE THREAD

**Issue by aleurcelay** [UBC-MDS/DSCI-532_group-211_dashboards#25]
Hello @rita-ni, @YueJiangMDSV & @arunmarria,
Remember you have to use Altair for this first dashboard. If you don't, you'll lose many marks for this! If you have more questions reach out to me :)

**Comment by arunmarria** [comment]
Hi @aleurcelay, thanks for the feedback.

But I am kind of confused here, As we already showed the dashboard  draft to 2 TAs so far in lab sessions. And nothing was brought up at that time or even before. And we were told  to use any tool to create dashboard provided we use dash and python. So we had an understanding of using python and dash to create a dashboard and not specifically altair to create dashboard. And for the second part it is R, I believe. And same was confirmed by TAs.

**Comment by YueJiangMDSV** [comment]
Yep, I remembered, we actually showed our dashboard to 3 TAs. None of them said anything. 
Plus, we also have a slider bar that shows the whole trend of the graph, which I do not think Altair can do.

**Comment by firasm** [comment]
Hi all, @aleurcelay asked me to step in here to clarify the situation.

Note that in both milestone 1 and milestone 2, it very clearly states that you need to use Altair  for your first app in python.

I'm not sure where this confusion arose from as we've been doing Altair with python, and ggplot2 with R since the start of DSCI 531. Could you let me know who told you to use "any tool" to create the dashboard?

Often it is very difficult to tell just by looking at a graph which library it came from and I only noticed it during the lab today because I caught a glimpse of some functionality that looked different and I'm pretty familiar with both Altair and plot.ly. Also, not all of the TAs in 532 were TAs in 531 so may not be able to recognize Altair specifically. Unless one of them specifically told you it was fine to use plot.ly for your plots, this reason will not fly.

<img width="930" alt="screen_shot_2019-11-28_at_11 40 08_pm" src="https://user-images.githubusercontent.com/2507459/69852436-2025b780-1239-11ea-9f1b-b9fabfe5b140.png">

<img width="721" alt="Screen Shot 2019-11-28 at 11 44 51 PM" src="https://user-images.githubusercontent.com/2507459/69852434-1ef48a80-1239-11ea-8b25-fcfcf5e19466.png">

Now - I understand it's pretty much Friday now and your apps are due in 36 hours or so and there likely will not be time for you to recreate all your plots in Altair. I can offer you three options:

1) Make all your plots using Altair.

2) You submit your app this week as-is using plot.ly with no penalty, but next week, you will need to  submit work for milestone 3 PLUS the dashboard you were supposed to submit this week using python and Altair. The next final dashboard (to be submitted at the end of week 4) will be using ggplotly so I expect you will have a head start  for that lab given that you've already created all your plots in plot.ly. 

3) Take a 30% penalty on the **viz** rubric (equivalent to 10% penalty on milestone2).

I suggest at this stage, you take option 2 or option 3. 

Let me know if you want to discuss this further tomorrow.

**Comment by YueJiangMDSV** [comment]
@firasm Hi, Firas, 
If we choose option 2 and commit to make the plots in Altair but turns we don’t have enough time in the coming week, will we still get option 3 as backup?

**Comment by arunmarria** [comment]
Thanks @firasm for clarifying and suggesting alternatives. I understand that by seeing plots it might be hard for TAs to tell the underlying tool. I re-read the instructions(screenshots you attached); I think it was our misinterpretation as a group as we took it more as python + dash exercise rather than altair +dash. I am not sure if 30% is adequate as it basically took us equivalent (or probably more) effort to learn and use plotly and dash. But we respect your decision. The only question we have as of now is the one pointed by @YueJiangMDSV, If we choose option 2 and commit to make the plots in Altair but turns we don’t have enough time(we have quizzes and labs next week) in the coming week, will we still get option 3 as backup ?

Thanks again!

**Comment by firasm** [comment]
hi @arunmarria , @rita-ni , @YueJiangMDSV - sorry for some reason I didn't get notified of your message in time. 

To answer your question: 

Yes, the default will be option 3 - we will deduct 30 from the viz mark. You can "remove" this penalty if you update your plots to be in Altair by the deadline of milestone3. 

I would wait until milestone3 is released before making a decision.

Please message me on Slack if you have any additional questions.

**Comment by arunmarria** [comment]
Hi @firasm , @aleurcelay 

As discussed we have updated our dash with **altair** plots. The submission details remain same and the app is deployed at same url but with altair plots. 

Thanks!

cc @YueJiangMDSV @rita-ni
